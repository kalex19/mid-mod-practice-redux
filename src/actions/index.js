export const addMusic = (music) => {
  type: 'ADD_MUSIC';
  payload: {
    music
  }
};

export const removeMusic = (id) => {
  type: 'REMOVE_MUSIC';
  payload: {
    id
  }
};

export const toggleFavorite = (favorite) => {
  type: 'TOGGLE_FAVORITE';
  payload: {
    favorite
  }
};