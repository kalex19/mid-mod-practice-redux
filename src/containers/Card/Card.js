import React, { Component } from 'react';
import { connect } from 'react-redux';
import { removeMusic } from '../../actions';

export class Card extends Component {
	render() {
		return (
			<div>
				<h2> {this.props.artist} </h2>
				<p>{this.props.song}</p>
				<button id="remove-btn" onClick={() => this.removeMusic(this.props.id)}>
					Delete Playlist Item
				</button>
			</div>
		);
	}
}

const mapDispatchToProps = dispatch => {
	removeMusic: id => dispatch(removeMusic(id));
};

export default connect(mapDispatchToProps)(Card);
