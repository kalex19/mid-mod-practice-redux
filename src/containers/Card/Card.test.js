import { Card, mapDispatchToProps } from './Card';
import { shallow } from 'enzyme';
import { removeMusic } from '../../actions';
import React from 'react';

const mockMusic = [
	{ artist: 'Frank', song: 'Sing to me', id: 1 },
	{ artist: 'Sinatra', song: 'All night long', id: 2 },
	{ artist: 'Brittany', song: 'Toxic', id: 3 }
];

describe('Card', () => {
	let wrapper, mockRemoveMusic;

	beforeEach(() => {
		mockRemoveMusic = jest.fn();
		wrapper.shallow(<Card {...mockMusic} removeMusic={mockRemoveMusic} />);
	});

	it('should mock snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should call removeMusic with id when button is clicked', () => {
		wrapper.find('#remove-btn').simulate('click');
		expect(mockRemoveMusic).toHaveBeenCalledWith(mockMusic.id);
	});

	describe('mapDispatchToProps', () => {
		it('should call dispatch with the correct action', () => {
			mockDispatch = jest.fn();
			const mockId = 2;
			const mappedProps = mapDispatchToProps(mockDispatch);
			const actionToProps = removeMusic(mockId);
			mappedProps.removeMusic(mockId);
			expect(mappedProps).toHaveBeenCalledWith(actionToProps);
		});
	});
});
