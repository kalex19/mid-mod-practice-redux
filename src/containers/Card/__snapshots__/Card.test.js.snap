// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should mock snapshot 1`] = `undefined`;
